linera-core/src/client/mod.rs

```diff
@@ -146,15 +146,54 @@ mod metrics {
     });
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+/// Defines what type of notifications we should process for a chain:
+/// - do we fully participate in consensus and download sender chains?
+/// - or do we only follow the chain's blocks without participating?
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum ListeningMode {
+    /// Listen to everything: all blocks for the chain and all blocks from sender chains,
+    /// and participate in rounds.
     FullChain,
+    /// Listen to all blocks for the chain, but don't download sender chain blocks or participate
+    /// in rounds. Use this when interested in the chain's state but not intending to propose
+    /// blocks (e.g., because we're not a chain owner).
+    FollowChain,
+}
+
+impl PartialOrd for ListeningMode {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        use std::cmp::Ordering;
+        match (self, other) {
+            (ListeningMode::FullChain, ListeningMode::FullChain) => Some(Ordering::Equal),
+            (ListeningMode::FullChain, _) => Some(Ordering::Greater),
+            (_, ListeningMode::FullChain) => Some(Ordering::Less),
+            (ListeningMode::FollowChain, ListeningMode::FollowChain) => Some(Ordering::Equal),
+        }
+    }
 }
```

Comment:
@afck Do we need to write this?

Reply:
Right, I guess on the testnet branch we could as well derive it, and make sure the variants are in the right order. But it's maybe more explicit this way, and on main we have to write it anyway because of the event sets?

Reply:
ok I was just curious